src/components/pages/AllAccommodation.js

```diff
@@ -41,14 +50,125 @@ export class AllAccommodation extends React.Component {
+        </Button>
+        <Modal
+          show={show}
+          onHide={() => setShow(false)}
+          size="lg"
+          aria-labelledby="example-custom-modal-styling-title"
+        >
+          <Modal.Header closeButton>
+            <Modal.Title id="example-custom-modal-styling-title">
+              Search Accommodations
+            </Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Form>
+              <Row className="center-items">
+                {
+                  (error.status) ? <AlertComponent variant="danger" heading={error.heading} message={error.message} /> : ''
+                }
+              </Row>
+              <Row>
+                <Col xs={12} sm={12} md={6} lg={6}>
+                  <Form.Group>
+                    <Form.Control as="select" name="location" data-test="location" onChange={this.handleChange}>
+                      <option selected disabled>Select Location...</option>
+                      <option value="">None</option>
+                      {
+                        locations ? locations.data.map((location) => <option key={location.id} value={location.name}>{location.name}</option>) : null
+                      }
+                    </Form.Control>
+                  </Form.Group>
+                </Col>
+                <Col xs={12} sm={12} md={6} lg={6}>
+                  <Form.Group>
+                    <Form.Control as="input" name="name" data-test="name" placeholder="Name" onChange={this.handleChange} />
+                  </Form.Group>
+                </Col>
+                <Col xs={12} sm={12} md={6} lg={6}>
+                  <Form.Group>
+                    <Form.Control as="input" name="description" data-test="description" placeholder="Description" onChange={this.handleChange} />
+                  </Form.Group>
+                </Col>
+                <Col xs={12} sm={12} md={6} lg={6}>
+                  <Form.Group>
+                    <Form.Control as="input" name="highlights" data-test="highlights" placeholder="Highlights" onChange={this.handleChange} />
+                  </Form.Group>
+                </Col>
+
+                <Col xs={12} sm={12} md={6} lg={6}>
+                  <Form.Group>
+                    <Form.Control as="input" name="amenities" data-test="amenities" placeholder="Amenities" onChange={this.handleChange} />
+                  </Form.Group>
+                </Col>
+                <Col xs={12} sm={12} md={6} lg={6}>
+                  <Form.Group>
+                    <Form.Control as="input" name="rating" data-test="rating" placeholder="Rating" onChange={this.handleChange} />
+                  </Form.Group>
+                </Col>
+              </Row>
+              <Row>
+                <Button className="full-width-buttons" data-test="submit-button" onClick={() => this.submitSearch()}>Search</Button>
+              </Row>
+            </Form>
+          </Modal.Body>
+        </Modal>
+      </Row>
+    );
```

Comment:
To avoid congesting this file, create a search component with all the search logic and call it here